src/libstd/error.rs

```diff
@@ -182,6 +182,13 @@ impl Error for string::FromUtf16Error {
     }
 }
 
+#[stable(feature = "str_parse_error2", since = "1.8.0")]
+impl Error for string::ParseError {
+    fn description(&self) -> &str {
+        "parse error"
```

Comment:
Ah here you can probably do the same `match *self {}` trick to avoid the need to return anything here at all.